sympy/solvers/tests/test_inequalities.py

```diff
@@ -323,15 +337,16 @@ def test_issue_10198():
         -1 + 1/abs(1/x - 1) < 0) == Or(
         And(-oo < x, x < 0), And(S(0) < x, x < S(1)/2)
         )
-    raises(NotImplementedError, lambda: reduce_inequalities(abs(1/sqrt(x)) - 1, x))
+    assert reduce_inequalities(abs(1/sqrt(x)) - 1, x) == Or(
+        And(-oo < x, x < 0), Eq(x, 1))
     assert reduce_abs_inequality(-3 + 1/abs(1 - 1/x), '<', x) == \
         Or(And(-oo < x, x < 0),
         And(S(0) < x, x < S(3)/4), And(S(3)/2 < x, x < oo))
     raises(ValueError,lambda: reduce_abs_inequality(-3 + 1/abs(
         1 - 1/sqrt(x)), '<', x))
 
 def test_issue_10047():
-    assert solve(sin(x) < 2) == And(-oo < x, x < oo)
+    assert solve(sin(x) < 2) == And(S(0) <= x, x < 2*pi)
```

Comment:
The whole test suite is passing so I think that the change in behaviour is backward compatible with `solve`. Since, we have followed (as well as expected) this behaviour throughout our test suite, I personally think that this test case can be modified in accordance with the restricted domain feature.

With regards to `solveset`, I was thinking that we can return the same result for now.
After the implementation of `BigUnion` and `BigIntersection`, we can extend the solution set (returned by `solve_univariate_inequality`) to return the complete solution.

What are your thoughts on this ?